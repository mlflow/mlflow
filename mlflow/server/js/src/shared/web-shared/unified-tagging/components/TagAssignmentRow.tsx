import invariant from 'invariant';
import React from 'react';

import { useDesignSystemTheme } from '@databricks/design-system';

export function TagAssignmentRow({ children }: { children: React.ReactNode }) {
  const { theme } = useDesignSystemTheme();

  const stableChildren = React.Children.toArray(children);
  invariant(stableChildren.length <= 3, 'TagAssignmentRow must have 3 children or less');

  const parsedChildren = Array(3)
    .fill(null)
    .map((_, i) => stableChildren[i] ?? <span key={i} style={{ width: theme.general.heightSm }} />); // Sync width with only icon button width

  return (
    <div css={{ display: 'grid', gridTemplateColumns: '1fr 1fr min-content', gap: theme.spacing.sm }}>
      {parsedChildren}
    </div>
  );
}
