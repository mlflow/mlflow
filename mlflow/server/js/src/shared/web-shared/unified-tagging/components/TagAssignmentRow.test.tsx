import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { TagAssignmentRow } from './TagAssignmentRow';

describe('TagAssignmentRow', () => {
  it('should throw an error if more than 3 children are passed', () => {
    const children = Array(4)
      .fill(null)
      .map((_, i) => <div key={i} />);

    const renderComponent = () =>
      render(
        <IntlProvider locale="en">
          <TagAssignmentRow>{children}</TagAssignmentRow>
        </IntlProvider>,
      );

    expect(renderComponent).toThrow('TagAssignmentRow must have 3 children or less');
  });

  it('should render children', () => {
    const children = Array(3)
      .fill(null)
      .map((_, i) => <div key={i} />);

    const { container } = render(
      <IntlProvider locale="en">
        <TagAssignmentRow>{children}</TagAssignmentRow>
      </IntlProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
