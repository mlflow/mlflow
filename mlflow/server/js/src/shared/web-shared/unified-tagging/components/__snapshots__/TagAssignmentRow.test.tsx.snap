// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TagAssignmentRow should render children 1`] = `
<div>
  <div
    css="[object Object]"
  >
    <div />
    <div />
    <div />
  </div>
</div>
`;
